system/src/system_publish_vitals.cpp

```diff
@@ -7,16 +7,33 @@
 #include "system_cloud.h"
 #include "system_threading.h"
 
-namespace {
+namespace
+{
 
 using namespace particle::system;
 
-int postDescription() {
-    const auto r = spark_protocol_post_description(spark_protocol_instance(), particle::protocol::DESCRIBE_METRICS, nullptr);
-    return spark_protocol_to_system_error(r);
+inline int postDescription()
```

Comment:
I would add a comment that this function is called from an ISR.

Reply:
Will do.

For posterity, it is called from more than just the ISR, but specifically it is `static` so it can be called from the ISR.

I will make a comment to reflect this.

Reply:
It's not called from an ISR though :)

Reply:
oh is it not? I thought it was called from an ISRTask

Reply:
Yeah, the name is somewhat confusing nowadays. Originally, `ISRTaskQueue` was used to process events posted from ISRs in the system thread, but now other threads use it too (mostly because there's no event queue available when threading is disabled)

Reply:
I've updated the comment to reflect the fact it is enqueued on the ISRTaskQueue, so it can be called on the system thread. :+1: